lib/src/component/ref_util.dart

```diff
@@ -394,19 +394,19 @@ UiFactory<TProps> Function(UiFactory<TProps>) forwardRef<TProps extends UiProps>
 ///       .._inputRef = ref
 ///     )();
 ///   },
-///   $Foo2Config, // ignore: undefined_identifier
+///   _$Foo2Config, // ignore: undefined_identifier
 /// );
 UiFactory<TProps> uiForwardRef<TProps extends bh.UiProps>(
-    dynamic Function(TProps props, dynamic ref) functionComponent, UiFactoryConfig<TProps> config) {
+    dynamic Function(TProps props, dynamic ref) functionComponent, dynamic config) {
```

Comment:
So that we don't have to deal with a whole bunch of dynamic calls that don't get any static analysis, can we create a casted version of the config?
```suggestion
    dynamic Function(TProps props, dynamic ref) functionComponent, dynamic _config) {
  
  final config = _config as UiFactoryConfig<TProps>;
```

We should probably also add an ArgumentError or assert before that, to provide an more useful error if consumers pass in the wrong type.

Also, can we document that `config` should always be a `UiFactoryConfig<TProps>`, and is just dynamic to avoid having to do a cast in the boilerplate?

(Same comment for uiFactory)